### PR TITLE
Add brief overview to readme

### PR DESCRIPTION
**Pull request by stephenworsley**
Add enough documentation to the readme so that new users should be able to make use of the most important features by following the example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,3 +7,60 @@
...
+# onto a cube with the same grid as `target_grid_cube`.
+result = regridder(source_mesh_cube)
+```
+
+Note that this pattern allows the reuse of an initialised regridder so long as the
+source and target grid/mesh are the same. Since initialisation often contains the
+most computationally intensive part of the regridding process, this pattern can save
+significant amounts of time when regridding. To make use of this efficiency across
+sessions, we support the saving of certain regridders (currently
+`MeshToGridESMFRegridder` and `GridToMeshESMFRegridder`). We can do this as follows:
```
Comment (trexfeathers) [inline README.md:53]: Don't bother listing which ones, you'll only have to change later.

File: README.md [hunk 2]
```diff
@@ -7,3 +7,60 @@
...
+
```
Comment (trexfeathers) [inline README.md:54]: I feel like we could get the same point across with half the characters. This is the first thing people see, let's make it snappy 😁

**Review by trexfeathers — CHANGES_REQUESTED** [review]
Thanks @stephenworsley, this does a good job of getting the point across. But we can make it shorter/easier; that's what I've seen when looking at other repos that rely on their `README` as the ice-breaker:

* https://github.com/actions/cache
* https://github.com/SciTools/cf-units

**Comment by trexfeathers** [comment]
Already looking loads better IMO 🙂

**Comment by zklaus** [comment]
I think the main path to regridding should still follow the standard approach in Iris for regridding as documented [here](https://scitools-iris.readthedocs.io/en/latest/userguide/interpolation_and_regridding.html#regridding), don't you think?

As such, the first and principal example should be something along the lines of
```
result = source_cube.regrid(target_cube, esmf_regrid.ESMFRegrid())
```

**Comment by stephenworsley** [comment]
@zklaus Ultimately, I think the documentation probably would benefit from looking more like the iris documentation. With that said, the current implementation of regridders does not go so far as to offer an exact match to iris functionality. This documentation reflects that and describes the regridding functionality which is common between the structured and unstructured regridders. The cached regridder is also necessary to introduce to discus regridder saving, which is a new feature not contained in iris. The unstructured and structured regridders are currently a little disjointed in their functionality and I hope to be unify them in a future release, however now that UGRID support in iris has released, I'm keen to release what we have for unstructured regridding as soon as possible, even if the structured and unstructured regridders do not yet have the same level of functionality.

**Review by trexfeathers — APPROVED** [review]

**Pull request merged**